packages/fast-tooling-react/src/data-utilities/mapping.ts

```diff
@@ -25,6 +25,113 @@ import {
 /**
  * This file contains all functionality for mapping data
  */
+export interface CodePreview {
+    componentName: string;
+    childOptions: CodePreviewChildOption[];
+    data: any;
+}
+
+export interface CodePreviewChildOption {
+    name: string;
+    schema: any;
+}
+
+export interface ReactChildren {
+    id: string;
+    propertyName: string;
+    location: string;
+    data: any;
+    component?: any;
+}
+
+export interface ReactComponentConfig {
+    hasChildren: boolean;
+    componentName: string;
+    componentData: any;
+    tab: string;
+    newTab: string;
+    location: string;
+}
+
+let variables: string;
+
+/**
+ * Maps data returned from the form generator to the React components
+ */
+export function mapDataToComponent(
+    schema: any,
+    data: any,
+    childOptions: ChildOptionItem[] = [],
+    plugins: Array<Plugin<PluginProps>> = []
+): MappedDataLocation[] {
+    const mappedData: any = cloneDeep(data);
+
+    // find locations of all items of data that are react children
+    let reactChildrenDataLocations: string[] = getDataLocationsOfChildren(
+        schema,
+        mappedData,
+        childOptions
+    );
+
+    // find locations of all items of data that are overridden by mappings
+    const pluginModifiedDataLocations: PluginLocation[] = getDataLocationsOfPlugins(
+        schema,
+        mappedData,
+        childOptions
+    );
+
+    // remove any children data locations from plugin modified locations
+    reactChildrenDataLocations = reactChildrenDataLocations.filter(
+        (reactChildrenDataLocation: string): string | undefined => {
+            if (
+                pluginModifiedDataLocations.findIndex(
+                    pluginFindIndexCallback(reactChildrenDataLocation)
+                ) === -1
+            ) {
+                return reactChildrenDataLocation;
+            }
+        }
+    );
+
+    // merge the plugin modified data locations with the children option data locations and categorize them
+    return reactChildrenDataLocations
+        .map(
+            (childDataLocation: string): MappedDataLocation => {
+                return {
+                    mappingType: DataResolverType.component,
+                    dataLocation: childDataLocation,
+                };
+            }
+        )
+        .concat(pluginModifiedDataLocations)
+        .sort(orderMappedDataByDataLocation)
+        .reduce((mappedDataReduced: any, mappedDataLocation: MappedDataLocation): any => {
+            return resolveData(
+                mappedDataLocation,
+                mappedDataReduced,
+                plugins,
+                childOptions
+            );
+        }, mappedData);
+}
+
+/**
+ * Maps data returned from the form generator to the code Preview
+ */
+export function mapDataToCodePreview(codePreview: CodePreview): string {
+    let cp: string = "TBD";
+    variables = "";
```

Comment:
Why are we assigning this an empty string and then concatenating it with the result of `generateReactCodePreview`? Doesn't that do nothing?